mirror-service/src/main/java/org/hyades/vulnmirror/datasource/nvd/NvdMirror.java

```diff
@@ -58,12 +67,22 @@ public Future<?> doMirror(String ecosystem) {
             } catch (Exception e) {
                 LOGGER.error("An unexpected error occurred mirroring the contents of the National Vulnerability Database", e);
                 dispatchNotification(LEVEL_ERROR, NOTIFICATION_TITLE,
-                        "An error occurred mirroring the contents of the National Vulnerability Database. Check log for details.");
+                        "An error occurred mirroring the contents of the National Vulnerability Database, cause being: "+e.getCause()+". Check log for details.");
             }
         });
     }
 
     void mirrorInternal() throws Exception {
+        Retry retry;
+        final RetryRegistry retryRegistry = RetryRegistry.of(RetryConfig.custom().
+                intervalFunction(ofExponentialBackoff(
+                        Duration.ofSeconds(config.retryBackoffInitialDurationSeconds()),
+                        config.retryBackoffMultiplier(), Duration.ofSeconds(config.retryMaxDuration())))
+                .maxAttempts(config.retryMaxAttempts())
+                .retryOnException(NvdApiException.class::isInstance)
+                .retryOnResult(response -> false)
+                .build());
+        retry = retryRegistry.retry("nvd-api");
```

Comment:
`retry` could be an `@ApplicationScoped` bean defined in `NvdMirrorConfiguration`.

Similar to how it's done in the vulnerability analyzer for circuit breakers:

https://github.com/DependencyTrack/hyades/blob/e90018987db42a8128b23424ed1ef04473974954/vulnerability-analyzer/src/main/java/org/hyades/processor/scanner/snyk/SnykProcessorConfiguration.java#L54-L70